internals-js/src/operations.ts

```diff
@@ -1535,6 +1547,53 @@ export class SelectionSet {
     this._selections = mapValues(keyedSelections);
   }
 
+  /**
+   * 
+   * @param namedFragments {NamedFragments}
+   * @param fragmentDefinitionsById 
+   * @returns 
+   */
+  minimizeSelectionSet(
+    namedFragments: NamedFragments = new NamedFragments(),
+    fragmentDefinitionsById: Map<number, [SelectionSet, NamedFragmentDefinition][]> = new Map(),
+  ): [SelectionSet, NamedFragments] {
+    const minimizedSelectionSet = this.lazyMap((selection) => {
+      if (selection.kind === 'FragmentSelection' && selection.element.typeCondition && selection.element.appliedDirectives.length === 0 && selection.selectionSet) {
+        const id = selection.key().length + selection.selectionSet.selections().length;
+        const equivalentSelectionSetCandidates = fragmentDefinitionsById.get(id);
+        if (equivalentSelectionSetCandidates) {
+          const match = equivalentSelectionSetCandidates.find(([candidateSet]) => candidateSet.equals(selection.selectionSet!));
+          if (match) {
+            return new FragmentSpreadSelection(this.parentType, namedFragments, match[1], []);
+          }
+        }
+
+        const [minimizedSelectionSet] = selection.selectionSet.minimizeSelectionSet(namedFragments, fragmentDefinitionsById);
+        const fragmentDefinition = new NamedFragmentDefinition(
+          this.parentType.schema(),
+          `__generated_on_${selection.element.typeCondition!.name}_${id}_${equivalentSelectionSetCandidates?.length ?? 0}`,
```

Comment:
I believe `__` is generally reserved for internal introspection stuff so I would avoid it.

Reply:
It's a valid `Name` and I do want to make it clear this is internal / implementation detail. I'll do 1 underscore or if you have something particular in mind lmk.

Reply:
yeah it is valid but might be confusing (and some subgraphs might be rejecting anything with `__` treating it as introspection) - single underscore is fine